src/InputGroup/InputGroup.js

```diff
@@ -1,272 +1,56 @@
-import Button from '../Button/Button';
 import classnames from 'classnames';
-import CustomPropTypes from '../utils/CustomPropTypes/CustomPropTypes';
-import FormInput from '../Forms/FormInput';
-import Icon from '../Icon/Icon';
+import InputGroupAddon from './_InputGroupAddon';
 import PropTypes from 'prop-types';
 import withStyles from '../utils/WithStyles/WithStyles';
-import { INPUT_GROUP_ADDON_POSITIONS, INPUT_GROUP_TYPES } from '../utils/constants';
 import React, { Component } from 'react';
 
 class InputGroup extends Component {
     constructor(props) {
         super(props);
-
-        this.state = {
-            value: this.props.inputValue,
-            searchValue: this.props.inputValue
-        };
     }
 
-    handleUp = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) + 1
-        });
-    };
-
-    handleDown = e => {
-        e.preventDefault();
-        this.setState({
-            value: parseInt(this.state.value, 10) - 1
-        });
-    };
-
-    handleClear = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: ''
-        });
-    };
-
-    handleChange = e => {
-        e.preventDefault();
-        this.setState({
-            searchValue: e.target.value
-        });
-    };
-
-    handleTextChange = e => {
-        e.preventDefault();
-        this.setState({
-            value: e.target.value
-        });
-    };
-
     render() {
         const {
-            actions,
-            addonClassNames,
-            addonPos,
-            addon,
             children,
             className,
             compact,
+            customStyles,
             disableStyles,
-            glyph,
-            inputClassName,
-            inputType,
-            inputId,
-            inputName,
-            inputPlaceholder,
-            inputProps,
-            inputValue,
-            localizedText,
-            numberDownButtonProps,
-            numberUpButtonProps,
             ...props
         } = this.props;
 
+        const inputClasses = 'fd-input-group__input';
+
         const inputGroupClasses = classnames(
             className,
             'fd-input-group'
         );
 
-        const addonClasses = classnames(
-            addonClassNames,
-            'fd-input-group__addon',
-            [{ 'fd-input-group__addon--button': !!actions || inputType === 'number' }]
-        );
-
-        const inputClasses = classnames(
-            inputClassName,
-            'fd-input-group__input'
+        return (
+            <div
+                {...props}
+                className={inputGroupClasses}>
+                {React.Children.map(children, (child) => (
+                    React.cloneElement(child, {
+                        compact,
+                        className: (child.type.displayname === InputGroup.Addon) ? '' : inputClasses
```

Comment:
I don't believe this comparison will ever match -- `InputGroup.Addon` is a function.  I think it should be this:
```jsx
className: (child.type.displayname === InputGroupAddon.displayName) ? '' : inputClasses
```

Reply:
`className: (child.type.displayname === InputGroupAddon.displayName) ? '' : inputClasses` does not work but className: (child.type.displayName === 'InputGroup.Addon') ? '' : inputClasses does 👍